apps/blog/_posts/bogosort.md

```diff
@@ -0,0 +1,104 @@
+
+Numa primeira travessia:
+
+- [**5**, **9**, 3, 4] → [**5**, **9**, 3, 4]
+  - Começamos por comparar os dois primeiros elementos e não os trocamos, visto que 5 < 9.
+- [5, **9**, **3**, 4] → [5, **3**, **9**, 4]
+  - De seguida, comparamos o 9 com o 3 e trocamos, já que 9 > 3.
+- [5, 3, **9**, **4**] → [5, 3, **4**, **9**]
+  - Por fim, comparamos o 9 com o 4 e também trocamos, visto que 9 > 4.
+
+Numa segunda travessia:
+
+- [**5**, **3**, 4, 9] → [**3**, **5**, 4, 9]
+  - Trocamos, já que 5 > 3.
+- [3, **5**, **4**, 9] → [3, **4**, **5**, 9]
+  - Trocamos, já que 5 < 4.
+- [3, 4, **5**, **9**] → [3, 4, **5**, **9**]
+  - Não trocamos.
+
+Conseguimos ver que no final da segunda travessia já temos a lista ordenada, mas o algoritmo ainda tem de fazer mais uma travessia para o saber.
+
+> **Aparte:** Reparaste que, a cada travessia, os maiores números vão agrupando-se no fundo da lista? Isto acontece porque o _bubble sort_ vai "empurrando" o maior número que encontra em cada iteração, sendo esse na primeira o 9 e na segunda o 5. Isto permite que seja feita uma pequena otimização no algoritmo: em cada iteração, pode-se deixar de verificar aqueles elementos que foram empurrados para o fundo da lista, porque já se sabe que estes não irão sair do sítio.
+
+## _Bogosort_
+
+Ao contrário do _bubble sort_, o _bogosort_ não possui uma sequência de passos que o leva a ordenar uma lista num período finito de tempo, baseando-se apenas no acaso.
+
+De que forma em concreto? Bem, é o equivalente a lançar um baralho de cartas ao ar, pegar nelas por uma ordem qualquer e verificar se estão por ordem. Se não estiverem, volta-se a repetir o processo. Devido a isto, não é considerado como um algoritmo propriamente útil para ordenação.
+
+## Notação _Big O_
+
+Não são necessárias medições para ver que o _bogosort_ é ineficiente, isto é, pode levar bastante tempo até conseguir ordenar uma lista.
+
+Coloca-se agora a questão: como é que podemos comparar dois algoritmos de forma objetiva para saber qual é o melhor?
+
+Existe na matemática uma notação conhecida como _Big O_, ou “Grande O” em português. Podemos utilizá-la para caracterizar a performance de um algoritmo relativamente ao número de vezes que atua sobre os elementos de uma lista, tendo por base o tamanho desta.
+
+Eis algumas das complexidades mais comuns, ordenadas por performance decrescente:
+
+- O(1) → diz-se que é **constante**, o algoritmo não varia consoante o tamanho do _input_ (num mundo ideal todos os algoritmos seriam assim, mas não é uma situação realista para ordenação)
+- O(log N) → diz-se que é **logarítmico**, o algoritmo cresce em complexidade a um ritmo mais lento que o aumento do tamanho do _input_
+- O(N) → diz-se que é **linear**, o algoritmo cresce em complexidade ao mesmo ritmo que o tamanho do _input_ aumenta
+- O(N^2) → diz-se que é **exponencial**, o algoritmo cresce em complexidade a um ritmo mais rápido que o aumento do tamanho do _input_
```

Comment:
This is not exponential. It's quadratic.

Reply:
My bad, you're right! I'll fix it right away :pray: